### DISCUSSION TITLE
AI + Qi + Lighthouse decks too heavy?

### DISCUSSION THREAD

**Discussion by machacekch** [bitcraze/discussions#1401] [Q&A] [answered]
I've been trying to fly these three decks at the same time, but the cf2 topples over immediately on taking off via the "Command Based Flight Control" in the cfclient. I've mounted the lighthouse deck above, and both the AI and Qi decks below. It flies just fine once the Qi deck is removed.

Any easy fixes I could try?

**Comment by knmcguire** [comment]
Hi! 

So I'm not sure if you found this page yet about the expansion decks in particular but it is handy to take a look at. https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/

The Qi + AIdeck + lighthouse deck is 5 + 4.4 + 2.7 =12.1 grams. The max. recommended payload weight is 15 g (see the specifications [here](https://www.bitcraze.io/products/crazyflie-2-1/)), however that is considered that they don't take up any consumption as well. The aideck is connected to the Vcom which is the supply also provided for the motors, so that doesn't leave much for the motor power to lift the full payload. If all the decks were just 'dead payloads' it would have been able to carry it, but since the lighthouse and aideck need consumption, the max payload is reduced. 

Another thing that you also have to check, is if the lighthouse deck is working together with the aideck. If you just hold the Crazyflie in your  hand without the motors engaged, can it still determine it's position? According to the deck -- deck table, the aideck and lighthouse are sharing a port.

I would advise you to take a look at the thrust upgrade kit to give your crazyflie some extra oomph to carry these decks: https://store.bitcraze.io/collections/bundles/products/thrust-upgrade-bundle-for-crazyflie-2-x

**Reply by machacekch** [reply]
Hi!

> So I'm not sure if you found this page yet about the expansion decks in particular but it is handy to take a look at. https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/
> 
> The Qi + AIdeck + lighthouse deck is 5 + 4.4 + 2.7 =12.1 grams. The max. recommended payload weight is 15 g (see the specifications [here](https://www.bitcraze.io/products/crazyflie-2-1/)), however that is considered that they don't take up any consumption as well.

Yes, I did the weight calculation beforehand and figured it should be fine. The AI deck isn't running anything at the moment, as I haven't managed to flash it yet (#1400), so it should be close to "dead weight".

> Another thing that you also have to check, is if the lighthouse deck is working together with the aideck. If you just hold the Crazyflie in your hand without the motors engaged, can it still determine it's position? According to the deck -- deck table, the aideck and lighthouse are sharing a port.

Yes, as stated above, once the Qi deck is removed it flies just fine with just the AI and LH decks. Even with all three decks the lighthouse localization works. Is the UART1 still used by the AI deck? [This doc](https://www.bitcraze.io/2023/03/ai-deck-documentation-and-examples-update/) suggests it communicates via NINA nowadays to free up UART1, but I may be misreading it.

> I would advise you to take a look at the thrust upgrade kit to give your crazyflie some extra oomph to carry these decks: https://store.bitcraze.io/collections/bundles/products/thrust-upgrade-bundle-for-crazyflie-2-x

Thanks, I'll consider adding it to my next order. However, if power draw is the limiting factor as you suggest, wouldn't these motors be just as underpowered?

**Reply by tobbeanton** [reply]
> Thanks, I'll consider adding it to my next order. However, if power draw is the limiting factor as you suggest, wouldn't these motors be just as underpowered?

I suggest you to order the 350mAh battery at the same time which can provide more power. I do believe the 250mAh battery will work as well but flight time will be very limited.

**Reply by machacekch** [reply, marked as answer]
I assembled another cf2 with the suggested components ([thrust upgrade kit](https://store.bitcraze.io/collections/bundles/products/thrust-upgrade-bundle-for-crazyflie-2-x) and [350mAh battery](https://store.bitcraze.io/products/350mah-lipo-battery)) and that one manages to carry all three decks.

Thanks!

**Reply by knmcguire** [reply]
Good to know! We will need to update that statement then because it seems very difficult to reach those 15 grams without the thrust upgrade kit.  However, if the aideck is powered, than it would probably have run the standard stock firmware, so it probably still would have drawn some power if the LEDs were running.

But you are all set now anyways :)
